File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/UserService.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.*;
import dev.typr.foundations.docs.core.UserRepo.User;

import java.sql.SQLException;
import java.util.List;
import java.util.Optional;

@SuppressWarnings("unused")
//start
public final class UserService {
    private final Transactor tx;

    public UserService(Transactor tx) {
        this.tx = tx;
    }

    public List<User> listUsers() throws SQLException {
        return UserRepo.selectAll.transact(tx);
    }

    public Optional<User> findUser(int id) throws SQLException {
        return UserRepo.selectById.on(id).transact(tx);
    }
}
//stop

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/UserService.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.docs.core.UserRepo.User

@SuppressWarnings(Array("unused"))
//start
class UserService(tx: Transactor):
  def listUsers(): List[User] =
    UserRepo.selectAll.transact(tx)

  def findUser(id: Int): Option[User] =
    UserRepo.selectById.on(id).transact(tx)
//stop
